File: tests/simple_setup_test_py/test_sample_class.cpp
#include "../../src/simple_setup_test_py_cpp/calc/SampleClass.h"
#include <catch2/catch_test_macros.hpp>

TEST_CASE("SampleClass behaves as expected", "[SampleClass]")
{
  simple_setup_test_py::calc::SampleClass sampleClass =
    simple_setup_test_py::calc::SampleClass();
  CHECK_NOTHROW(sampleClass.setValue(10));
  REQUIRE(sampleClass.getValue() == 10);
  REQUIRE(sampleClass.getValue() != 11);
}

File: tests/simple_setup_test_py/test_sample_class.py
import unittest

from simple_setup_test_py_cpp import SampleClass


class SampleClassTestCase(unittest.TestCase):
    def test_sample_class(self):
        sample_obj = SampleClass()
        self.assertIsInstance(sample_obj, SampleClass)
        for i in range(10):
            sample_obj.set_value(1)
            self.assertEqual(sample_obj.get_value(), 1)
